### ISSUE TITLE
Index silently got corrupted

### ISSUE THREAD

**Issue by johny65** [orientechnologies/orientdb#9161]
### OS: Ubuntu 19.10

I have an index on an integer property. In some moment, it got corrupted; when I do a query using the index, the same vertex (same RID) is returned twice:

![Captura de pantalla de 2020-03-05 08-19-27](https://user-images.githubusercontent.com/5256776/75977846-04772080-5ebc-11ea-9aa2-ed42902ab170.png)

After rebuilding the index, the query returns the correct result:

![Captura de pantalla de 2020-03-05 08-19-35](https://user-images.githubusercontent.com/5256776/75977912-22448580-5ebc-11ea-96ae-90d1255020a5.png)

It's not the first time that my indexes got corrupted, the problem is that I have no way of knowing that, only when I see incorrect results of a query I'm doing myself. In this case, I discovered the problem because an edge was created twice between two vertices:

```
let $a = select from #some:rid;
let $b = select from V where indexed_property = some_value; /* this returned the same vertex twice */
create edge from $a to $b; /* as $b is a collection with the same vertex repeated, the edge is created twice between the vertices */

**Comment by laa** [comment]
HI @johny65 , what type of index do you use?

**Comment by johny65** [comment]
@laa: NOTUNIQUE, CELL_BTREE.

**Comment by laa** [comment]
@johny65 we fixed similar issue in 3.0.30 version, vertex was deleted but the related record was not, could you try 3.0.30 and check that you do not have problems any more?

**Comment by laa** [comment]
Also to prevent usage of old version of indexes, I suggest you export a database to JSON then import it or create and refill from scratch.

**Comment by johny65** [comment]
> @johny65 we fixed similar issue in 3.0.30 version, vertex was deleted but the related index record was not, could you try 3.0.30 and check that you do not have problems any more?

Ok, when 3.0.30 is out I'll check if it happens again.


> Also to prevent usage of old version of indexes, I suggest you export a database to JSON then import it or create and refill from scratch.

I exported the database to a Gzip, but when import it fails with:

```
Error: com.orientechnologies.orient.core.exception.OStorageException: Error sending import request

Error: com.orientechnologies.orient.core.db.tool.ODatabaseExportException: Error on importing database 'db' from file: /tmp/import13938435091701872541db.gz

Error: com.orientechnologies.orient.core.db.tool.ODatabaseImportException: Error on importing record

Error: com.orientechnologies.orient.core.exception.OSerializationException: Error on unmarshalling JSON content for record #225:0

Error: java.lang.IllegalArgumentException: Impossible to add a null identifiable in a ridbag
```

**Comment by laa** [comment]
@johny65 could you send me resulted JSON, will check it.

**Comment by laa** [comment]
It is fixed in the latest 3.0.x. We will release it ASAP.
